internals/cli/cli.go

```diff
@@ -63,52 +63,34 @@ var optionsData options
 // ErrExtraArgs is returned  if extra arguments to a command are found
 var ErrExtraArgs = fmt.Errorf("too many arguments for command")
 
-// cmdInfo holds information needed to call parser.AddCommand(...).
-type cmdInfo struct {
-	name, shortHelp, longHelp string
-	builder                   func() flags.Commander
-	hidden                    bool
-	optDescs                  map[string]string
-	argDescs                  []argDesc
-	alias                     string
-	extra                     func(*flags.Command)
+// CmdInfo holds information needed to call parser.AddCommand(...).
+type CmdInfo struct {
+	Name, ShortHelp, LongHelp string
+	Builder                   func() flags.Commander
+	Hidden                    bool
+	OptDescs                  map[string]string
+	ArgDescs                  []argDesc
```

Comment:
Per preview review, the goal here is not to just make things public, but to actually polish them enough so that making it public is reasonable. OptDesc and ArgDesc are not good names, and are close enough together that it's hard to see why they both exist. Even more when they are undocumented. Also, we should pretty much never have public types exposing public types as is the case of ArgDescs above, since clients won't be able to define those values.